sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuGetJsonObject.scala

```diff
@@ -83,25 +84,33 @@ private[this] object JsonPathParser extends RegexParsers {
     }
   }
 
-  def containsUnsupportedPath(instructions: List[PathInstruction]): Boolean = {
-    // Gpu GetJsonObject is not supported if JSON path contains wildcard [*]
-    // see https://github.com/NVIDIA/spark-rapids/issues/10216
-    instructions.exists {
-      case Wildcard => true
-      case Named(name) if name == "*" => true
-      case _ => false
+  def fallbackCheck(instructions: List[PathInstruction]): Boolean = {
+    // JNI kernel has a limit of 16 nested nodes, fallback to CPU if we exceed that
+    instructions.length > 16
```

Comment:
JNI now is using 32

Reply:
done but maybe needs more test on it. Will test: If kernel works good under 32, and will plugin fallback above 32.

Reply:
done